@shared/api/internal.ts

```diff
@@ -183,32 +183,40 @@ export const confirmPassword = async (
   return response;
 };
 
-export const getAccountBalance = async (
+export const getAccountDetails = async (
   publicKey: string,
-): Promise<{
-  balance: string;
-  isFunded: boolean;
-}> => {
-  let response = {
-    balances: [{ asset_code: "", balance: "" }],
-    isFunded: true,
-  };
+): Promise<AccountDetailsInterface> => {
+  const dataProvider = new DataProvider({
+    serverUrl: NETWORK_URL,
+    accountOrKey: publicKey,
+    networkPassphrase: NETWORK_PASSPHRASE,
+  });
+
+  let payments = null;
+  let balances = null;
+  let isFunded = null;
+
+  try {
+    ({ balances } = await dataProvider.fetchAccountDetails());
+    const transactionData = await dataProvider.fetchPayments({
+      limit: TRANSACTIONS_LIMIT,
+    });
+    payments = transactionData?.records || {};
+  } catch (e) {
+    console.error(e);
+  }
 
   try {
-    response = await server.loadAccount(publicKey);
+    isFunded = await dataProvider.isAccountFunded();
   } catch (e) {
+    isFunded = false;
     console.error(e);
-    // TODO: Check that this is indeed a 404 before returning 0
-    return { balance: "0", isFunded: false };
   }
```

Comment:
using a separate try/catch here as I don't want an issue with `fetchAccountDetails` or `fetchPayments` to erroneously flag an account as "unfunded" and give someone a heart attack